common/src/main/java/com/terraformersmc/cinderscapes/mixin/MixinServerWorld.java

```diff
@@ -36,33 +38,38 @@ protected MixinServerWorld(MutableWorldProperties properties, RegistryKey<World>
      */
     @Inject(method="tickIceAndSnow", at = @At(value = "HEAD"), locals = LocalCapture.NO_CAPTURE)
     private void cinderscapes$tickAsh(BlockPos tickPos, CallbackInfo ci) {
+        // Ashy shoals only exists in the nether, why do this iteration on any other dimension
+        if (!getDimensionEntry().matchesKey(DimensionTypes.THE_NETHER)) return;
         if (CinderscapesConfig.INSTANCE.enableAshFall) {
             BlockPos pos = tickPos.mutableCopy();
-            BlockState state = getBlockState(pos);
-            RegistryEntry<Biome> biome = this.getBiome(pos);
 
-            while (pos.getY() < 127 && !(
-                    biome.matchesKey(CinderscapesBiomes.ASHY_SHOALS) &&
-                    state.isSideSolidFullSquare(this, pos, Direction.UP) &&
-                    blockAbove(pos).isIn(CinderscapesBlockTags.ASH_PERMEABLE) &&
-                    this.getBlockState(pos.up()).isAir() &&
-                    CinderscapesBlocks.ASH.getDefaultState().canPlaceAt(this, pos.up()))) {
-                pos = pos.up();
-                state = getBlockState(pos);
-                biome = this.getBiome(pos);
-            }
-
-            if (pos.getY() < 127) {
-                this.setBlockState(pos.up(), CinderscapesBlocks.ASH.getDefaultState());
+            for (; pos.getY() < 127; pos.setY(pos.getY() + 1)) {
```

Comment:
The `BlockPos.Mutable#move` method would better indicate the intent here:

```suggestion
            for (; pos.getY() < 127; pos.move(Direction.UP)) {
```

Reply:
move has a lot of operations that are meaningless 100% of the time in this particular calculation:
![image](https://github.com/TerraformersMC/Cinderscapes/assets/72876796/924e03c3-4328-4ae2-aabc-d11039affd32)

setY does exactly one thing, the one thing that's needed in this situation.

Reply:
I'm not sure if the additional field lookups on a JVM level would be prohibitively expensive in this scenario, especially when there are larger optimizations that could be done. I would wait until the other optimizations are implemented and then benchmark `setY` versus `move` to see if using `setY` is worthwhile.